tests/unit/test_printer.py

```diff
@@ -1123,3 +1123,100 @@ def test_spinner_silent_on_complete_messages(spinner, monkeypatch):
     time.sleep(0.05)
 
     assert spinner.printer.spinned == []
+
+
+def test_secrets(capsys, log_filepath):
```

Comment:
It could be interesting to add tests for secrets not matching word boundaries (e.g. "term" and "range" to get "apple banana o***** wa*****elon"), multiple repetitions of the secret in the same string, or non-ASCII unicode characters.

Reply:
these are all very good suggestions, thanks! Added some more tests